packages/lib/src/vis/tiles/hooks.ts

```diff
@@ -1,13 +1,32 @@
-import { useThree } from '@react-three/fiber';
+import type { Domain } from '@h5web/shared';
+import { scaleLinear } from '@visx/scale';
 
 import { useVisibleDomains } from '../hooks';
+import { useAxisSystemContext } from '../shared/AxisSystemContext';
 
-export function useDataPerPixel() {
+export function useScaledVisibleDomains(
+  xRange: Domain,
+  yRange: Domain
+): {
+  xVisibleDomain: Domain;
+  yVisibleDomain: Domain;
+} {
   const { xVisibleDomain, yVisibleDomain } = useVisibleDomains();
-  const { width, height } = useThree((state) => state.size);
+  const { abscissaConfig, ordinateConfig } = useAxisSystemContext();
+
+  const xScale = scaleLinear({
+    domain: abscissaConfig.visDomain,
+    range: xRange,
+    clamp: true,
+  });
+  const yScale = scaleLinear({
+    domain: ordinateConfig.visDomain,
+    range: yRange,
+    clamp: true,
+  });
 
   return {
-    xDataPerPixel: (xVisibleDomain[1] - xVisibleDomain[0]) / width,
-    yDataPerPixel: (yVisibleDomain[1] - yVisibleDomain[0]) / height,
+    xVisibleDomain: xVisibleDomain.map(xScale) as Domain,
+    yVisibleDomain: yVisibleDomain.map(yScale) as Domain,
```

Comment:
Is using same naming for output as `useVisibleDomains` ok for you or best use a different naming (e.g., `xScaledVisibleDomain`... but that starts to be long)?

Reply:
I think it's fine.

Reply:
Agreed.